### PR TITLE
file_input uses a reasonable default for max_concurrent_files

### PR DESCRIPTION
**Pull request by djaglowski**
## **Please check that the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Add a changelog entry (for non-trivial bug fixes / features)
- [ ] CI passes

### CODE REVIEW

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.5000546|-0.034436345|127.724945|-1.0408173|
|1|5000|5.0345483|-0.10353184|138.13443|+0.021820068|
|1|10000|10.345002|+0.17221832|144.03934|-0.8076477|
|1|50000|49.98382|-5.066799|171.18723|-0.94667053|
|1|100000|106.01578|-3.6419754|245.30334|-3.4971619|
|10|100|1.9828336|+0.25869453|134.03354|+0.64601135|
|10|500|6.775978|+0.74148417|140.29431|+2.0787964|
|10|1000|11.879361|-1.7245035|149.3416|+2.3935852|
|10|5000|57.696472|+2.317276|180.33904|+1.4684753|
|10|10000|110.42671|-8.425774|226.45798|-10.451233|

**Comment by djaglowski** [comment]
This will make it impossible for the user to specify "unlimited" by explicitly setting `max_concurrent_files: 0`. However, it's not clear that this behavior is really ever desired anyways. If so, it can be approximated by setting `max_concurrent_files: HUGE_NUMBER`.

File: operator/builtin/input/file/config.go [hunk 1]
```diff
@@ -90,7 +90,9 @@ func (c InputConfig) Build(context operator.BuildContext) ([]operator.Operator,
 		return nil, fmt.Errorf("`max_log_size` must be positive")
 	}
 
-	if c.MaxConcurrentFiles <= 1 {
+	if c.MaxConcurrentFiles == 0 {
+		c.MaxConcurrentFiles = defaultMaxConcurrentFiles
```
Comment (jsirianni) [inline operator/builtin/input/file/config.go:94]: I think `NewInputConfig` already sets this to 1024. I added print statements to a build (from master) and verified that default is 1024, and if I set it to 10 (or any other value), it is respected.

In fact, with 40k files in a directory, setting to 10 prevents "too many open files". I think we might want to choose something less than 1024, maybe 1000 (this works on my test instance).

**Review by jsirianni — CHANGES_REQUESTED** [review]

File: operator/builtin/input/file/config.go (hunk 1 above)
Comment (djaglowski) [reply operator/builtin/input/file/config.go:94]: Ah, good point. I'll close this PR.

Regarding the default though - you're saying 1000 works but 1024 does not? Seems like an unsubstantial change. Does 1000 come from anywhere in particular?

Comment (jsirianni) [reply operator/builtin/input/file/config.go:94]: On most systems, 1024 will be the default limit for a process (I believe). Stanza already has a file handle open to a couple things by default. When file_input demands 1024, we hit the limit earlier than expected.

Comment (djaglowski) [reply operator/builtin/input/file/config.go:94]: Makes sense. Should we give it more margin than 24 then? Maybe set it to 512?

Comment (jsirianni) [reply operator/builtin/input/file/config.go:94]: I think 512 is more than reasonable. Anything higher, the user should really be defining their upper limit.

**Pull request closed without merging**